mmv1/third_party/terraform/services/compute/resource_compute_instance_template_test.go.erb

```diff
@@ -1235,6 +1235,100 @@ func TestAccComputeInstanceTemplate_NetworkAttachment(t *testing.T) {
 }
 <% end -%>
 
+func TestAccComputeInstanceTemplate_migration(t *testing.T) {
+	acctest.SkipIfVcr(t)
+	t.Parallel()
+
+	var instanceTemplate compute.InstanceTemplate
+	suffix := acctest.RandString(t, 10)
+	oldVersion := map[string]resource.ExternalProvider{
+		"google": {
+			VersionConstraint: "4.84.0", // a version that doesn't separate user defined labels and system labels
+			Source:            "registry.terraform.io/hashicorp/google",
+		},
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		CheckDestroy:             testAccCheckComputeInstanceTemplateDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config:            testAccComputeInstanceTemplate_basic(suffix),
```

Comment:
You can write a check func that stores the numeric id (`id` server-side). We could use two of those to compare the resources and confirm we didn't recreate it!

Reply:
(same w/ the test below)